File: HTTP_Server_CLI/scenarios/3.1.1-middleware_3.lua
-- @label: 3 middleware routes
-- @group: Middleware routes
-- @competitors: bootgly
-- Round-robins through protected routes that apply middleware.

local paths = {"/protected/dashboard", "/protected/settings", "/protected/profile"}
local counter = 0

request = function()
   counter = counter + 1
   return wrk.format("GET", paths[(counter % #paths) + 1])
end

File: HTTP_Server_CLI/scenarios/php/3.1.1-middleware_3.php
<?php
// @label: 3 middleware routes
// @group: Middleware routes
// @competitors: bootgly
// Round-robins through protected routes that apply middleware.

return [
   'method' => 'GET',
   'paths'  => ['/protected/dashboard', '/protected/settings', '/protected/profile'],
];
